pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_allocation_handlers.go

```diff
@@ -911,20 +922,37 @@ func (p *DynamicPolicy) generatePoolsAndIsolation(poolsQuantityMap map[string]in
 		proportionalPoolsQuantityMap := make(map[string]int)
 
 		for poolName, poolQuantity := range poolsQuantityMap {
```

Comment:
can those logic be refined (for instance, put in another function)? kind of too large to understand ...